src/System.Runtime.Extensions/ref/project.json

```diff
@@ -1,6 +1,7 @@
 {
   "dependencies": {
-    "System.Runtime": "4.4.0-beta-24629-01"
+    "System.Runtime": "4.4.0-beta-24629-01",
+    "System.Security.Principal": "4.4.0-beta-24629-01"
```

Comment:
Dang. Thanks for point out that System.Threading.Thread depends on System.Security.Principal. For now given that System.Security.Principal has minimal dependencies we can keep this dependency and move forward with what you are doing. System.Security.Principal will end up being a candidate for elimination when we look at all the dependencies after all the porting work is done.